app/services/lara_serialization_helper.rb

```diff
@@ -0,0 +1,53 @@
+class LaraSerializationHelper
+
+  def initialize
+    @entries = {}
+  end
+
+  def cache_item(ref_id, item)
+    @entries[ref_id] = item
+  end
+
+  def cache_item_copy(original_item, copy_of_item)
+    cache_item(key(original_item), copy_of_item)
+  end
+
+  def lookup_item(key_sting)
+    @entries[key_sting]
+  end
+
+  def lookup_new_item(original_item)
+    lookup_item(key(original_item))
+  end
+
+  def key(interactive)
+    "#{interactive.id}-#{interactive.class.name}"
+  end
```

Comment:
Since this now seems generic `interactive` should be renamed. Perhaps `item` to follow the argument names below.